paperqa/agents/helpers.py

```diff
@@ -0,0 +1,238 @@
+from __future__ import annotations
+
+import logging
+import os
+import re
+from datetime import datetime
+from typing import Any, cast
+
+from anthropic import AsyncAnthropic
+from openai import AsyncOpenAI
+from rich.table import Table
+
+from .. import (
+    AnthropicLLMModel,
+    Docs,
+    OpenAILLMModel,
+    embedding_model_factory,
+    llm_model_factory,
+)
+from ..llms import LangchainLLMModel
+from .models import AnswerResponse, QueryRequest
+
+logger = logging.getLogger(__name__)
+
+
+def get_year(ts: datetime | None = None) -> str:
+    """Get the year from the input datetime, otherwise using the current datetime."""
+    if ts is None:
+        ts = datetime.now()
+    return ts.strftime("%Y")
+
+
+async def openai_get_search_query(
+    question: str,
+    count: int,
+    template: str | None = None,
+    llm: str = "gpt-4o-mini",
+    temperature: float = 1.0,
+) -> list[str]:
+    if isinstance(template, str):
+        if not (
+            "{count}" in template and "{question}" in template and "{date}" in template
+        ):
+            logger.warning(
+                "Template does not contain {count}, {question} and {date} variables. Ignoring template."
+            )
+            template = None
+
+        else:
+            # partial formatting
+            search_prompt = template.replace("{date}", get_year())
+
+    if template is None:
+        search_prompt = (
+            "We want to answer the following question: {question} \n"
+            "Provide {count} unique keyword searches (one search per line) and year ranges "
+            "that will find papers to help answer the question. "
+            "Do not use boolean operators. "
+            "Make sure not to repeat searches without changing the keywords or year ranges. "
+            "Make some searches broad and some narrow. "
+            "Use this format: [keyword search], [start year]-[end year]. "
+            "where end year is optional. "
+            f"The current year is {get_year()}\n\n"
```

Comment:
Should we have trailing `\n\n` here? Usually callers join together prompts as they see fit

Disclaimer: I know this is how it's done internally, but maybe it's not a good pattern

Reply:
Sure we can cut this